pkg/resources/model/reconciler.go

```diff
@@ -276,8 +276,9 @@ func NewValidationReconciler(
 			if req.IsEmptyPatch() {
 				continue
 			}
+			obj := req.Obj.DeepCopyObject()
```

Comment:
could move the type assertion here, where it's more obviously valid (since `req.Obj` is a `client.Object` by definition, thus the copy will be too...)
```suggestion
			obj := req.Obj.DeepCopyObject().(client.Object)
```